internal/scaffold/resource.go

```diff
@@ -101,6 +102,55 @@ func (r *Resource) Validate() error {
 	return nil
 }
 
+// ResourceData holds data used to construct a Resource.
+type ResourceData struct {
+	APIVersion string
+	Domain     string
+	Group      string
+	Version    string
+	Kind       string
+}
+
+// ToResource transforms ResourceData into a full Resource.
+func (d ResourceData) ToResource() (Resource, error) {
+	if err := d.validate(); err != nil {
+		return Resource{}, fmt.Errorf("invalid resource: %v", err)
+	}
+
+	apiVersion := d.APIVersion
+	if apiVersion == "" {
+		apiVersion = fmt.Sprintf("%s.%s/%s", d.Group, d.Domain, d.Version)
+	}
+	r, err := NewResource(apiVersion, d.Kind)
+	return *r, err
+}
+
+// Validate ensures fields in ResourceData conform to their corresponding Kubernetes specs.
+func (d ResourceData) validate() error {
+	if d.Domain == "" {
+		return errors.New("domain cannot be empty")
+	}
+	if d.Group == "" {
+		return errors.New("group cannot be empty")
+	}
+	if strings.HasSuffix(d.Group, d.Domain) {
+		return fmt.Errorf("group %q cannot contain a domain suffix", d.Group)
+	}
```

Comment:
Do we want to check if the domain and group name form a "qualified name", i.e. "validation.IsQualifiedName"?